tests/x11/nis_client.pm

```diff
@@ -17,19 +17,31 @@ use testapi;
 use mm_network;
 use lockapi;
 use utils qw(systemctl turn_off_gnome_screensaver);
+use version_utils 'is_sle';
 
 sub run {
     my ($self) = @_;
     x11_start_program('xterm -geometry 155x45+5+5', target_match => 'xterm');
-    turn_off_gnome_screensaver if check_var('DESKTOP', 'gnome');
     become_root;
+    turn_off_gnome_screensaver if check_var('DESKTOP', 'gnome');
     configure_default_gateway;
     configure_static_ip('10.0.2.3/24');
     configure_static_dns(get_host_resolv_conf());
+    if ($self->firewall eq 'firewalld') {
```

Comment:
Let's add check if service is already defined, like we have in yast2_nis.pm:
```perl
if ($self->firewall eq 'firewalld' && script_run 'firewall-offline-cmd --get-services | grep ypbind') {
```
and extract common part, as we do same trick, but by manually adding service with firewall-cmd

Reply:
ypbind is being installed after start of "yast2 nis" on line 40, actually that was the initial reason why was the job failing in openQA. Therefore I have introduced a needle to check if there is any package to be installed by yast2 by itself. I guess, that the additional condition will be always true at this point.